application/src/main/java/org/thingsboard/server/service/entitiy/ota/DefaultTbOtaPackageService.java

```diff
@@ -64,12 +67,12 @@ public OtaPackageInfo save(SaveOtaPackageInfoRequest saveOtaPackageInfoRequest,
 
     @Override
     public OtaPackageInfo saveOtaPackageData(OtaPackageInfo otaPackageInfo, String checksum, ChecksumAlgorithm checksumAlgorithm,
-                                             byte[] data, String filename, String contentType, User user) throws ThingsboardException {
+                                             MultipartFile file, User user) {
         TenantId tenantId = otaPackageInfo.getTenantId();
         OtaPackageId otaPackageId = otaPackageInfo.getId();
         try {
             if (StringUtils.isEmpty(checksum)) {
-                checksum = otaPackageService.generateChecksum(checksumAlgorithm, ByteBuffer.wrap(data));
+                checksum = ChecksumUtil.generateChecksum(checksumAlgorithm, file.getInputStream());
```

Comment:
Multiple calls to file.getInputStream is not good, right?

Reply:
probably yes? But its impossible to reuse same FileInputStream several times